extensions/amp-image-slider/0.0/amp-image-slider.js

```diff
@@ -0,0 +1,566 @@
+      this.element.addEventListener('touchend', this.handleTapImage);
+      dev().assertElement(this.barButton_)
+          .addEventListener('mousedown', this.dragStart);
+      dev().assertElement(this.barButton_)
+          .addEventListener('touchstart', this.touchStart);
+    }
+
+    this.isEventRegistered_ = true;
+  }
+
+  /**
+   * Unregister event handlers
+   */
+  unregisterEvents() {
+    if (this.isHoverSlider_) {
+      this.element.removeEventListener('mousemove', this.handleHideHint, true);
+      this.container_.removeEventListener('mousemove', this.handleHover);
+    } else {
+      this.element.removeEventListener('mousedown', this.handleHideHint, true);
+      this.element.removeEventListener('touchstart', this.handleHideHint, true);
+      this.element.removeEventListener('click', this.handleClickImage);
+      this.element.removeEventListener('touchend', this.handleTapImage);
+      dev().assertElement(this.barButton_)
+          .removeEventListener('mousedown', this.dragStart);
+      dev().assertElement(this.barButton_)
+          .removeEventListener('touchstart', this.touchStart);
+      // remove pointer related events below
+      this.dragEnd(null);
+      this.touchEnd(null);
+    }
+
+    this.isEventRegistered_ = false;
+  }
+
+  /**
+   * Handle hover event
+   * @param {Event} e
+   */
+  handleHover(e) {
+    // This offsetWidth may change if user resize window
+    // Thus not cached
+    // const {left, width} = this.getLayoutBox();
+    const {left, width} = this.container_./*OK*/getBoundingClientRect();
+    const leftPercentage = (e.pageX - left) / width;
+    this.updatePositions(leftPercentage);
+  }
+
+  /**
+   * Handle click on the image
+   * @param {Event} e
+   */
+  handleClickImage(e) {
+    // const {left, width} = this.getLayoutBox();
+    const {left, width} = this.container_./*OK*/getBoundingClientRect();
+    const leftPercentage = (e.pageX - left) / width;
+    this.animateUpdatePositions(leftPercentage);
+  }
+
+  /**
+   * Handle tap on the image
+   * @param {Event} e
+   */
+  handleTapImage(e) {
+    // const {left, width} = this.getLayoutBox();
+    const {left, width} = this.container_./*OK*/getBoundingClientRect();
+    if (e.touches.length > 0) {
+      const leftPercentage = (e.touches[0].pageX - left) / width;
+      this.animateUpdatePositions(leftPercentage);
+    }
+  }
+
+  /**
+   * Handle hinding the hint
+   * @param {Event} unusedEvent
+   */
+  handleHideHint(unusedEvent) {
+    if (!this.isBarHintHidden_ && this.barHint_) {
+      this.isBarHintHidden_ = true;
+      this.barHint_.classList.add('i-amphtml-image-slider-hint-hidden');
+    }
+  }
+
+  /**
+   * Animated, update element positions based on percentage
+   * @param {number} toPercentage
+   */
+  animateUpdatePositions(toPercentage) {
+    const {left: containerLeft, width: containerWidth}
+    //    = this.getLayoutBox();
+        = this.container_./*OK*/getBoundingClientRect();
```

Comment:
Calls to `getBoundingClientRect()` should be wrapped in a `this.measureElement` because it's expensive. Is this a value that we can cache and only update when necessary? Ideally, the dimensions + positions of your container should only change on `onMeasureChanged`.